jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```diff
@@ -669,8 +680,33 @@ protected void createTempDirectory()
                 // Delete and recreate it to ensure it is empty
                 if (tempDirectory.exists() && !IO.delete(tempDirectory))
                     throw new IllegalArgumentException("Failed to delete temp dir: " + tempDirectory);
-                if (!tempDirectory.mkdirs())
-                    throw new IllegalArgumentException("Unable to create temp dir: " + tempDirectory);
+                try
+                {
+                    Files.createDirectories(tempDirectory.toPath());
+                }
+                catch (IOException e)
+                {
+                    Path path = tempDirectory.toPath();
+                    while (!"/".equals(path.toString()))
+                    {
+                        System.out.println("Path '" + path + "' exists? " + Files.exists(path));
+                        if (Files.exists(path))
+                        {
+                            try
+                            {
+                                UserPrincipal owner = Files.getOwner(path);
+                                Set<PosixFilePermission> posixFilePermissions = Files.getPosixFilePermissions(path);
+                                System.out.println("  user name: " + System.getProperty("user.name") + " file owner: " + owner + " perm: " + posixFilePermissions);
+                            }
+                            catch (Exception ex)
+                            {
+                                ex.printStackTrace();
+                            }
+                        }
+                        path = path.getParent();
+                    }
+                    throw new IllegalArgumentException("Unable to create temp dir: " + tempDirectory, e);
```

Comment:
I see that you are using `System.out` here. Doesn't it make more sense to use the logger? I was also comparing this a bit with the way you are creating the temp directory in `WebInfConfiguration`

https://github.com/jetty/jetty.project/blob/55e307224782b4b7ed4cc0756deafdd492c9417c/jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/WebInfConfiguration.java#L155-L182

A slightly different approach is used there. e.g. `Files.createTempDirectory` from Java itself. 

Again, I have no clue about the codebase, and I'm just comparing with a subset of things I can see.

Reply:
That was very temporary to help me troubleshoot some test failures in CI. Now that I've figured them out, I reverted that ugly change.

Reply:
oops sorry for the noise in that case. I won't mix in your review anymore. Thank you very much for looking into my issue and working on a fix for it.